Controllers/articles-controller.js

```diff
@@ -29,4 +29,31 @@ function getArticles(req, res, next){
     });
 }
 
-module.exports = {getArticleByID, getArticles};
\ No newline at end of file
+function getCommentsByID(req, res, next) {
+    const { article_id } = req.params;
+
+    // Promise.all([fetchArticleByID(article_id), fetchCommentsByID(article_id)])
+    // .then((comments)=>{
+
+    //     res.status(200).send(comments)
+    // })
+    // .catch((err)=>{
+    //     next(err);
+    // });
+
+    fetchArticleByID(article_id)
+    .then(()=>{
+        fetchCommentsByID(article_id)
+        .then((comments)=>{
+            res.status(200).send(comments)
+        })
+        .catch((err)=>{
+            next(err);
+        }); 
+    })
+    .catch((err)=>{
+        next(err);
+    });
```

Comment:
Hey Luke, you're really close with both methods ( _I can see you were trying Promise.all above_ ) 

If you have a look at the notes page on Promises [https://notes.northcoders.com/courses/js-fundamentals/promises](url) in the section titled **Benefits of Using Promises** there is an example of a promise chain using _.then_ blocks .